File: backend/src/main/java/com/amazonclone/amazonclonebackend/entities/Product.java
package com.amazonclone.amazonclonebackend.entities;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import net.minidev.json.annotate.JsonIgnore;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;


@Getter
@Setter
@EqualsAndHashCode
@ToString
@Entity
@Table(name = "product", schema = "orders")
public class Product implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;
    private String name;
    private String description;
    private Float price;
    @Column(name = "quantity")
    private Integer quantity;
    @Column(name = "barcode", nullable = false, updatable = false)
    private String barcode;
    private String imageUrl;
    private Integer ratings;

    @JsonManagedReference(value = "product")
    @OneToMany(targetEntity = ProductInOrdination.class, mappedBy = "product",cascade = CascadeType.MERGE)
    @JsonIgnore
    @ToString.Exclude
    private List<ProductInOrdination> productInOrdination;

}

File: frontend/src/app/Components/products/product.ts
export class Product {

  id!: null;
  name!: string;
  description!: string;
  price!: number;
  quantity!:number;
  barcode!: null;
  imageUrl!: string;
  ratings!: number;

  constructor(id:null, name: string, desc: string, price: number,quantity:number, barcode:null,imageUrl:string,ratings:number){
    this.id =id;
    this.name = name;
    this.description = desc;
    this.price = price;
    this.quantity=quantity;
    this.barcode = barcode;
    this.imageUrl = imageUrl;
    this.ratings = ratings;
  }




}
